Library/Homebrew/cmd/shellenv.sh

```diff
@@ -18,7 +20,7 @@ homebrew-shellenv() {
     return
   fi
 
-  case "$(/bin/ps -p "${PPID}" -c -o comm=)" in
+  case "${1:-$(/bin/ps -p "${PPID}" -c -o comm=)}" in
```

Comment:
I think this is a bit too simple a fix. If you pass multiple parameters the 2nd onwards will be silently ignored. It's also not obvious from scanning this file that the `1` here is the use of argument.

Can I suggest:
- pulling the argument into a named variable and using that instead e.g. `HOEMBREW_SHELL_NAME=$1` or something
- doing some basic checks for too many arguments being provided and throwing an error (perhaps overkill, maybe check what other `.sh` commands do here.
- alternatively, pass a single argument from `brew.sh` rather than all arguments

Reply:
I did both.

I debated, in the `[[ -n "$1" ]]` case, whether we should check for exactly the shells listed at the top and *warn* if there’s no match (e.g., `echo >&2 "shell $1 is not a supported shell, falling back to sh"`). I decided against because the actual logic is to fall back to sh formatting regardless.